akka-persistence-query/src/main/scala/akka/persistence/query/javadsl/DurableStateStoreQuery.scala

```diff
@@ -30,13 +30,16 @@ trait DurableStateStoreQuery[A] extends DurableStateStore[A] {
    * This will return changes that occurred up to when the `Source` returned by this call is materialized. Changes to
    * objects made since materialization are not guaranteed to be included in the results.
    *
+   * The [[DurableStateChange]] elements can be [[akka.persistence.query.UpdatedDurableState]] or `DeletedDurableState`.
+   * `DeletedDurableState` is not implemented yet, see issue https://github.com/akka/akka/issues/30446.
+   *
    * @param tag The tag to get changes for.
    * @param offset The offset to get changes since. Must either be [[akka.persistence.query.NoOffset]] to get
    *               changes since the beginning of time, or an offset that has been previously returned by this query.
    *               Any other offsets are invalid.
    * @return A source of change in state.
    */
-  def currentChanges(tag: String, offset: Offset): Source[DurableStateChange[A], NotUsed]
+  def currentChanges(tag: String, offset: Offset): Source[DurableStateChange, NotUsed]
```

Comment:
Hmm, in this API it was *kinda* nice to get the generic so you knew the `value` of the update would be of that type.

(it's not very meaningful for the 'deleted' case, of course, but perhaps doesn't harm much either?)

Reply:
ok, so I added the type parameter to DurableStateChange even if it's not used? For the delete class it doesn't have to be included.

Reply:
You can now get rid of the `@unchecked` in `PersistenceTestKitDurableStateStoreSpec.scala`, right?

Reply:
Good catch. I didn't know that it would also make that better.